src/naive_bayes/gaussian.rs

```diff
@@ -425,6 +425,23 @@ mod tests {
         );
     }
 
+    #[test]
+    fn run_gaussian_naive_bayes_with_few_samples() {
+        let x =
+            DenseMatrix::<f64>::from_2d_array(&[&[-1., -1.], &[-2., -1.], &[-3., -2.], &[1., 1.]]);
+        let y: Vec<u32> = vec![1, 1, 1, 2];
+
+        let gnb = GaussianNB::fit(&x, &y, Default::default());
+
+        match gnb.unwrap().predict(&x) {
+            Ok(_) => assert!(false, "test should return Failed"),
+            Err(err) => {
+                assert!(err.to_string() == "Can't find solution: log_likelihood for distribution of one of the rows is NaN");
```

Comment:
is this the same behavior if we run this with `scikit`?

Reply:
```
>>> x = [[-1., -1.], [-2., -1.], [-3., -2.], [1., 1.]]
>>> y = [1, 1,1,2]
>>> from sklearn.naive_bayes import GaussianNB
>>> clf = GaussianNB()
>>> clf.fit(x, y)
GaussianNB()
>>> clf.predict(x)
array([1, 1, 1, 2])
```

Seems to work for me, maybe we need to check instead what is done there to avoid the issue. I guess that it is related to negative input but need to check

Reply:
yea every library has its own policy about dealing with `NaN`. i agree we should try to follow `sklearn` but that example seems just like returning `y` to me, so it's hard to say what the general behaviour should be.